x-pack/metricbeat/module/statsd/server/server.go

```diff
@@ -191,13 +192,30 @@ func (m *MetricSet) getEvents() []*mb.Event {
 	return events
 }
 
+// ServerStart starts the underlying m.server
+func (m *MetricSet) ServerStart() {
+	if m.serverStarted {
```

Comment:
I forgot to set `m.serverStarted` to true/false indeed, fixed now

`ServerStart()`/`ServerStop()` are called only in the test and here: https://github.com/elastic/beats/pull/29543/files#diff-e6af89a723439b5852afec015561c45d9f424aa0d68746f7ed899f948b8109eaR219-R220

the reason for adding `m.serverStarted` is indeed to avoid starting the server twice from the test.
we can add an extra mutex in case to guard `m.serverStarted`